### PR TITLE
Adds optional props offsetX offsetY frameHeight and frameWidth

### PR DESCRIPTION
**Pull request by genesy**
For my use case I had a spritesheet with multiple sprite animation and sizes. I didn't want to manually crop the sprites so I just resorted to adding these optional params which allowed me to define the shape of the sprites and where the frames should start from.

### CODE REVIEW

**Comment by MiLeung** [comment]
Can you update the README so that it explains how to use these new props?

**Comment by genesy** [comment]
updated

File: src/SpriteSheet.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { Animated, Easing, View } from "react-native";
+import Svg,{Image}from 'react-native-svg';
```
Comment (MiLeung) [inline src/SpriteSheet.js:2]: What's this for?

File: src/SpriteSheet.js [hunk 2]
```diff
@@ -50,13 +53,15 @@ export default class SpriteSheet extends React.PureComponent {
...
-    let { source, height, width, rows, columns } = this.props;
+    let { source, height, width, rows, columns, frameHeight, frameWidth, offsetY = 0, offsetX = 0 } = this.props;
     let image = resolveAssetSource(source);
     let ratio = 1;
     let imageHeight = image.height;
     let imageWidth = image.width;
-    let frameHeight = image.height / rows;
-    let frameWidth = image.width / columns;
+    offsetX = -offsetX;
+    offsetY = -offsetY;
```
Comment (MiLeung) [inline src/SpriteSheet.js:62]: Why not just mention in the docs that offsetX and offsetY have to be negative or positive?

Comment (genesy) [reply src/SpriteSheet.js:62]: Having it act like css offsets seems more natural imo

File: src/SpriteSheet.js (hunk 1 above)
Comment (genesy) [reply src/SpriteSheet.js:2]: must've missed this when i was testing a different approach to making images not pixelated when enlarging. will remove.

File: src/SpriteSheet.js (hunk 2 above)
Comment (MiLeung) [reply src/SpriteSheet.js:60]: fair

**Comment by MiLeung** [comment]
Published in v1.1.10

File: src/SpriteSheet.js [hunk 3]
```diff
@@ -50,13 +51,15 @@ export default class SpriteSheet extends React.PureComponent {
     this.time = new Animated.Value(0);
     this.interpolationRanges = {};
 
-    let { source, height, width, rows, columns } = this.props;
+    let { source, height, width, rows, columns, frameHeight, frameWidth, offsetY = 0, offsetX = 0 } = this.props;
```
Comment (MiLeung) [inline src/SpriteSheet.js:54]: I moved `offsetY = 0, offsetX = 0` to `defaultPropTypes` in v1.1.10

**Comment by MiLeung** [comment]
Can you confirm this bug and maybe make a fix to it? https://github.com/MiLeung/rn-sprite-sheet/issues/27

**Pull request merged**